src/sprints/sprints.service.ts

```diff
@@ -273,91 +280,159 @@ export class SprintsService {
-                    notes,
-                },
-            });
-            return updatedMeeting;
-        } catch (e) {
-            if (e.code === "P2025") {
-                throw new NotFoundException(`Invalid meetingId: ${meetingId}`);
-            }
+        const meeting = await this.prisma.teamMeeting.findUnique({
+            where: {
+                id: meetingId,
+            },
+            select: {
+                voyageTeamId: true,
+            },
+        });
+        if (!meeting) {
+            throw new NotFoundException(
+                `Meeting with Id ${meetingId} does not exist.`,
+            );
         }
+        manageOwnVoyageTeamWithIdParam(req.user, meeting.voyageTeamId);
+        const updatedMeeting = await this.prisma.teamMeeting.update({
+            where: {
+                id: meetingId,
+            },
+            data: {
+                title,
+                description,
+                meetingLink,
+                dateTime,
+                notes,
+            },
+        });
+        return updatedMeeting;
     }
 
     async createMeetingAgenda(
         meetingId: number,
         { title, description, status }: CreateAgendaDto,
+        req: CustomRequest,
     ) {
-        try {
-            const newAgenda = await this.prisma.agenda.create({
-                data: {
-                    teamMeetingId: meetingId,
-                    title,
-                    description,
-                    status,
-                },
-            });
-            return newAgenda;
-        } catch (e) {
-            if (e.code === "P2003") {
-                throw new BadRequestException(
-                    `Invalid meetingId: ${meetingId}`,
-                );
-            }
+        const meeting = await this.prisma.teamMeeting.findUnique({
+            where: {
+                id: meetingId,
+            },
+            select: {
+                voyageTeamId: true,
+            },
+        });
+
+        if (!meeting) {
+            throw new NotFoundException(
+                `Meeting with Id ${meetingId} does not exist.`,
+            );
         }
+
+        manageOwnVoyageTeamWithIdParam(req.user, meeting.voyageTeamId);
+
+        const newAgenda = await this.prisma.agenda.create({
+            data: {
+                teamMeetingId: meetingId,
+                title,
+                description,
+                status,
+            },
+        });
+        return newAgenda;
     }
 
     async updateMeetingAgenda(
         agendaId: number,
         { title, description, status }: UpdateAgendaDto,
+        req: CustomRequest,
     ) {
-        try {
-            const updatedMeeting = await this.prisma.agenda.update({
-                where: {
-                    id: agendaId,
-                },
-                data: {
-                    title,
-                    description,
-                    status,
+        const agenda = await this.prisma.agenda.findUnique({
```

Comment:
This is a good solution even though we added one extra database query. I'm not sure if there's another way to just have 1 query with the addition of authorisation